src/Projection/HttplugProjectionManager.php

```diff
@@ -0,0 +1,294 @@
+<?php
+/**
+ * This file is part of the prooph/httplug-event-store.
+ * (c) 2017-2017 prooph software GmbH <contact@prooph.de>
+ * (c) 2017-2017 Sascha-Oliver Prolic <saschaprolic@googlemail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Prooph\EventStore\Httplug\Projection;
+
+use Http\Client\HttpClient;
+use Http\Discovery\MessageFactoryDiscovery;
+use Http\Message\RequestFactory;
+use Prooph\EventStore\Exception\ProjectionNotFound;
+use Prooph\EventStore\Exception\RuntimeException;
+use Prooph\EventStore\Httplug\Exception\NotAllowed;
+use Prooph\EventStore\Projection\ProjectionManager;
+use Prooph\EventStore\Projection\ProjectionStatus;
+use Prooph\EventStore\Projection\Projector;
+use Prooph\EventStore\Projection\Query;
+use Prooph\EventStore\Projection\ReadModel;
+use Prooph\EventStore\Projection\ReadModelProjector;
+
+final class HttplugProjectionManager implements ProjectionManager
+{
+    /**
+     * @var HttpClient
+     */
+    private $httpClient;
+
+    /**
+     * @var RequestFactory
+     */
+    private $requestFactory;
+
+    public function __construct(
+        HttpClient $httpClient,
+        RequestFactory $requestFactory = null
+    ) {
+        $this->httpClient = $httpClient;
+        $this->requestFactory = $requestFactory ?: MessageFactoryDiscovery::find();
+    }
+
+    public function createQuery(): Query
+    {
+        throw new \BadMethodCallException(__METHOD__ . ' not implemented');
```

Comment:
Will this be possible in future versions?

I'm asking because the use case is very interesting. If service A publishes its event stream via http api, service B could consume it using http-lug-event-store with a projection

Reply:
That would be nice, indeed. But I am not yet sure on how this can be implemented.

Reply:
Maybe we should rethink projections one more time. I plan an experiment once the new arangoDB event store is ready (with the native php extension developed by @sandrokeil and @martin-schilling). You can add stream processing capabilities to arango similar to the mongodb 3.6 stream watch feature.

The streaming API provides a way to start at a specific position in the stream and consume messages from that position forward (maybe we can use http/2 to establish a permanent connection). If a new message is written to the stream it is pushed to the consumer as long as the consumer watches the stream.

The consumer on the other side uses a local store to remember current stream position. 
Maybe server and consumer need to run within an event loop. With react/http supporting PSR-7 now this becomes a very interesting field to explore.